py-smart-open: Adding new version and dependencies

marcusboden opened the pull request:
My first a bit more complex dependency situation. Feel free to comment/corret.
The dependency handling changed quite a lot with newer versions of smart-open. I've decided to include all variants by default.

Commit:
py-smart-open: Adding new version and dependencies

adamjstewart commented on var/spack/repos/builtin/packages/py-smart-open/package.py:
```diff
@@ -13,14 +13,48 @@ class PySmartOpen(PythonPackage):
-    depends_on('py-bz2file', when='^python@:2', type=('build', 'run'))
+
+    with when('@5:'):
+        depends_on('python@3.6:', type=('build', 'run'))
```
```suggestion
        depends_on('python@3.6:3', type=('build', 'run'))
```

adamjstewart commented on var/spack/repos/builtin/packages/py-smart-open/package.py:
```diff
@@ -13,14 +13,48 @@ class PySmartOpen(PythonPackage):
+
+        # google cloud support
+        variant(
+            'gcs', default=True, description='Builds a shared version of the library'
```
Descriptions need to be updated

adamjstewart reviewed (changes requested):
I would recommend making all variants False by default. This minimizes the number of dependencies that are built. Also, it's unlikely that someone will be using the same installation on both AWS and GCP.

Commit:
fix python version and variant description/default

marcusboden commented:
I've updated the descriptions and changed the defaults for gcs, s3 and azure. But I kept the http as default, as I think that http is a pretty standard usecase and it only need py-requests.
